### ISSUE TITLE
several tests failing in pypy3

### ISSUE THREAD

**Issue by asottile** [asottile/pyupgrade#602]
I suspect a regression there but I have not had time to triage this

```
=========================== short test summary info ============================
FAILED tests/features/dict_literals_test.py::test_dictcomps[dict((a, b) for a, b in y)-{a: b for a, b in y}]
FAILED tests/features/dict_literals_test.py::test_dictcomps[dict((a, b,) for a, b in y)-{a: b for a, b in y}]
FAILED tests/features/dict_literals_test.py::test_dictcomps[dict((a, b, ) for a, b in y)-{a: b for a, b in y}]
FAILED tests/features/dict_literals_test.py::test_dictcomps[dict([a, b] for a, b in y)-{a: b for a, b in y}]
FAILED tests/features/dict_literals_test.py::test_dictcomps[dict([(a, b), c] for a, b, c in y)-{(a, b): c for a, b, c in y}]
FAILED tests/features/dict_literals_test.py::test_dictcomps[dict(((a), b) for a, b in y)-{(a): b for a, b in y}]
FAILED tests/features/dict_literals_test.py::test_dictcomps[dict((k, dict((k2, v2) for k2, v2 in y2)) for k, y2 in y)-{k: {k2: v2 for k2, v2 in y2} for k, y2 in y}]
FAILED tests/features/dict_literals_test.py::test_dictcomps[dict((a, b)for a, b in y)-{a: b for a, b in y}]
FAILED tests/features/dict_literals_test.py::test_dictcomps[dict(\n    (\n        a,\n        b,\n    )\n    for a, b in y\n)-{\n        a:\n        b\n    for a, b in y\n}]
FAILED tests/features/dict_literals_test.py::test_dictcomps[x(\n    dict(\n        (a, b) for a, b in y\n    )\n)-x(\n    {\n        a: b for a, b in y\n    }\n)]
FAILED tests/features/set_literals_test.py::test_sets[set((a, b) for a, b in y)-{(a, b) for a, b in y}]
================== 11 failed, 976 passed, 5 xfailed in 48.93s ==================
ERROR: InvocationError for command /home/asottile/workspace/pyupgrade/.tox/pypy3/bin/coverage run -m pytest tests (exited with code 1)
```

**Comment by asottile** [comment]
this appears to be the main difference -- which is probably a bug in pypy:

```diff
$ diff -u <(.tox/py38/bin/astpretty t.py) <(.tox/pypy3/bin/astpretty t.py)
--- /dev/fd/63	2022-02-15 18:20:18.314171145 -0500
+++ /dev/fd/62	2022-02-15 18:20:18.314171145 -0500
@@ -14,9 +14,9 @@
                 args=[
                     GeneratorExp(
                         lineno=1,
-                        col_offset=3,
+                        col_offset=4,
                         end_lineno=1,
-                        end_col_offset=25,
+                        end_col_offset=24,
                         elt=Tuple(
                             lineno=1,
                             col_offset=4,
@@ -34,7 +34,7 @@
                                     lineno=1,
                                     col_offset=15,
                                     end_lineno=1,
-                                    end_col_offset=19,
+                                    end_col_offset=16,
                                     elts=[
                                         Name(lineno=1, col_offset=15, end_lineno=1, end_col_offset=16, id='a', ctx=Store()),
                                         Name(lineno=1, col_offset=18, end_lineno=1, end_col_offset=19, id='b', ctx=Store()),
$ cat t.py
set((a, b) for a, b in y)
```

**Comment by asottile** [comment]
that's the 3.7 behaviour -- so probably the pypy parser didn't update for the 3.8 changes

**Comment by asottile** [comment]
this appears to have been a bug in pypy3, fixed with 7.3.8
